ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -817,7 +834,33 @@ func (r *RayClusterReconciler) SetupWithManager(mgr ctrl.Manager, reconcileConcu
 			predicate.LabelChangedPredicate{},
 			predicate.AnnotationChangedPredicate{},
 		))).
-		Watches(&source.Kind{Type: &corev1.Event{}}, &handler.EnqueueRequestForObject{}).
+		Watches(&source.Kind{Type: &corev1.Event{}},
+			&handler.EnqueueRequestForObject{},
+			builder.WithPredicates(predicate.Funcs{
+				CreateFunc: func(e event.CreateEvent) bool {
+					// get the v1.Event object
```

Comment:
nit: The comment above this code snippet appears to be unnecessary as the code is self-explanatory and can be easily understood without additional comments.
```suggestion
```